MyOwnList/MyList.cs

```diff
@@ -61,285 +38,426 @@ public MyList(MyList<T> collection)
+
+        public MyList()
         {
+            Length = 0;
+            _array = new T[8];
+        }
 
-            if (!IsValidCapacity(Count))
+        public MyList(T val)
+        {
+            Length = 0;
+            _array = new T[8];
+            Add(val);
+        }
+
+        public MyList(MyList<T> collection)
+        {
+            Length = 0;
+            _array = new T[8];
+            AddRange(collection);
+        }
+
+        public void Add(T item)
+        {
+            if (!IsValidCapacity(Length))
             {
                 Resize();
             }
 
-            array[Count] = item;
-            ++Count;
+            _array[Length] = item;
+            ++Length;
         }
 
         public void Clear()
         {
-            Count = 0;
-            array = new T[8];
+            Length = 0;
+            _array = new T[8];
         }
 
         public override string ToString()
         {
-            string s = "";
+            StringBuilder stringBuilder = new StringBuilder();
 
-            for (int i = 0; i < Count; i++)
+            for (int i = 0; i < Length; i++)
             {
-                s += array[i];
+                stringBuilder.Append(_array[i]).Append(" ");
             }
-            return s;
+
+            return stringBuilder.ToString().Trim();
         }
 
         public T[] ToArray()
         {
-            T[] arrayNew = new T[Count];
-            for (int i = 0; i < Count; i++)
+            T[] arrayNew = new T[Length];
+
+            for (int i = 0; i < Length; i++)
             {
-                arrayNew[i] = array[i];
+                arrayNew[i] = _array[i];
             }
+
             return arrayNew;
         }
 
-        public void AddPos(int pos, T val)
+        public void AddByIndex(int index, T value)
         {
-            if (IsValidCount(pos))
+            ++Length;
+            if (IsValidLength(index))
             {
-                ++Count;
-                if (Count >= Capacity)
+                Resize();
+
+                for (int i = Length - 1; i > index; i--)
                 {
-                    Resize();
+                    Swap(ref _array[i], ref _array[i - 1]);
                 }
-                for (int i = Count - 1; i > pos; i--)
-                {
 
-                    array[i] = array[i - 1];
+                _array[index] = value;
+            }
+            else
+            {
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        public void AddStart(T value)
+        {
+            int index = 0;
+            AddByIndex(index, value);
+        }
 
+        public void AddRangeByIndex(int index, MyList<T> collection)
+        {
+            int count = collection.Count();
+
+            Length += count;
+
+            if (IsValidLength(index))
+            {
+                Resize();
+
+                for (int i = Length - 1; i > index; i--)
+                {
+                    if (i >= count)
```

Comment:
You can move it to `for` declaration `i > index && i >= count`
Maybe you can even optimize it